Source/WebCore/platform/encryptedmedia/CDMProxy.cpp

```diff
@@ -235,6 +237,16 @@ void CDMProxy::updateKeyStore(const KeyStore& newKeyStore)
 {
     Locker locker { m_keysLock };
     m_keyStore.merge(newKeyStore);
+
+    for (auto& keyHandle : newKeyStore) {
+        std::unique_ptr promise = m_keyHandlePromises.take(keyHandle->id());
```

Comment:
if you didn't take the producer from the m_keyHandlePromises you could just resolve it if it exists and is `true` (Producer::operator bool() ) 

unless the value of a key can change over time (but in which case things would be broken anyway)